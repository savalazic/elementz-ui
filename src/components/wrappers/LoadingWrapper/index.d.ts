export { default, LoadingWrapperProps } from './LoadingWrapper';
