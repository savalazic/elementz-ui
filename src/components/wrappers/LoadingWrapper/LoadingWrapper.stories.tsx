import React from 'react';

import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import { infoConfig } from '../../../utils/info-config';

import { LoadingWrapper } from './LoadingWrapper';
import { Spinner } from '../../elements';

const stories = storiesOf('Wrappers|LoadingWrapper', module);

stories.addParameters(infoConfig);

stories
  .add('default', () => {
    const loading = boolean('Is loading', true);

    return (
      <LoadingWrapper isLoading={loading}>
        {({ isLoading }) => (
          <div>
            <p>
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ducimus
              temporibus placeat excepturi ex provident fugit ratione ab saepe
              culpa, repellendus unde explicabo repellat error reprehenderit
              consequatur quas eveniet alias quis?
            </p>
            {isLoading && <Spinner />}
          </div>
        )}
      </LoadingWrapper>
    );
  })
  .add('example usage with overlay', () => {
    const loading = boolean('Is loading', true);

    const wrapperStyles: any = {
      position: 'relative',
      backgroundColor: loading ? 'rgba(0, 0, 0, 0.6)' : 'transparent',
    };

    const loaderStyles: any = {
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
      color: 'white',
    };

    return (
      <LoadingWrapper isLoading={loading}>
        {({ isLoading }) => (
          <div style={wrapperStyles}>
            <p>
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Ducimus
              temporibus placeat excepturi ex provident fugit ratione ab saepe
              culpa, repellendus unde explicabo repellat error reprehenderit
              consequatur quas eveniet alias quis?
            </p>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Cum quas
              laboriosam enim aut officiis in amet hic at illum, quidem aliquid,
              accusantium a perspiciatis libero veritatis sed nobis voluptate
              doloribus dicta odio laudantium quis eligendi. Eaque officia, hic
              facilis, odit architecto minima modi fugit fuga vitae porro sequi
              molestiae iure?
            </p>
            {isLoading && (
              <div style={loaderStyles}>
                <Spinner />
                <p>loading...</p>
              </div>
            )}
          </div>
        )}
      </LoadingWrapper>
    );
  });
