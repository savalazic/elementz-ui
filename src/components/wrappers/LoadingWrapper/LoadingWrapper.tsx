import React, { Component } from 'react';
import styled from 'styled-components';
import { Spinner } from '../../elements';

export interface LoadingWrapperProps {
  /** loading state */
  isLoading: boolean;
  /** children */
  children: (params: { isLoading: boolean }) => React.ReactNode;
}

const StyledLoadingWrapper = styled.div<LoadingWrapperProps>``;

export class LoadingWrapper extends Component<LoadingWrapperProps> {
  public state = {
    isLoading: this.props.isLoading,
  };

  public render() {
    return this.props.children({
      isLoading: this.state.isLoading,
    });
    // return (
    //   <StyledLoadingWrapper isLoading={this.state.isLoading}>
    //     {this.props.children}
    //   </StyledLoadingWrapper>
    // );
  }
}

// export const LoadingWrapper = ({
//   children,
//   isLoading,
// }: LoadingWrapperProps) => {
//   return (
//     <StyledLoadingWrapper isLoading={isLoading}>
//       <Spinner visible={isLoading} />
//       {children}
//     </StyledLoadingWrapper>
//   );
// };

export default LoadingWrapper;
