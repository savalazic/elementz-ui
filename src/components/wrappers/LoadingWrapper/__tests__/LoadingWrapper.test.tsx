import 'jest-dom/extend-expect';
import React from 'react';
import { render } from 'react-testing-library';

import { LoadingWrapper } from '../LoadingWrapper';

describe('LoadingWrapper component', () => {
  it('should match default snapshot', () => {
    const { container } = render(
      <LoadingWrapper isLoading>
        {({ isLoading }) => <div>Loading..</div>}
      </LoadingWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
