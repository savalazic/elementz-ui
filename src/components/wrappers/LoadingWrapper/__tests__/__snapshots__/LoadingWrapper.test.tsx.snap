// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWrapper component should match default snapshot 1`] = `
<div>
  <div>
    Loading..
  </div>
</div>
`;
